scripts/cargo-rmc

```diff
@@ -94,6 +63,134 @@ def main():
+        
+        return parser
+
+    # Check for conflicting flags
+    def validate(args):
+        rmc.ensure(not (args.crate and args.crate_flag), "Please provide a single crate to verify.")   
+        rmc.ensure(not (args.no_config_toml and args.config_toml), "Incompatible flags: --config-toml, --no-config-toml")
+
+    # Fix up args before returning
+    def post_process(args):
+        # Combine positional and flag argument for input
+        cl_args.crate = cl_args.crate or cl_args.crate_flag or "."
+
+        # --quiet overrides --verbose
+        if args.quiet:
+            args.verbose = False
+
+        # Add some CBMC flags by default unless `--no-default-checks` is being used
+        if args.default_checks:
+            rmc.add_selected_default_cbmc_flags(args)
+
+    # Combine args set by config toml and provided on command line
+    def get_combined_args(config_toml, crate, cl_args):
+        # Check that config toml exists, or set to default (<crate>/Config.toml)
+        if config_toml is not None:
+            rmc.ensure(pathlib.Path(config_toml).is_file(), f"Invalid config file: {config_toml}")
+        else:
+            crate_cargo_toml = pathlib.Path(crate).joinpath("Cargo.toml")
+            if crate_cargo_toml.is_file():
+                config_toml = crate_cargo_toml
+            else:
+                return cl_args
+        
+        # Extract flags from config toml
+        toml_args = load_toml(config_toml)
+
+        # Parse args from both toml and command line
+        # but let command line args override toml args
+        combined_args = argparse.Namespace()
+        parser.parse_args(toml_args, namespace=combined_args)
+        parser.parse_args(namespace=combined_args)
+
+        return combined_args
+
+    # Remove "rmc" from arg if invoked `cargo rmc ...`
+    if len(sys.argv) >= 2 and sys.argv[1] == "rmc":
+        del sys.argv[1]
+
+    # Parse command line args to find crate and check for config flags
+    parser = create_parser()
+    cl_args = parser.parse_args()
+    validate(cl_args)
+
+    # Load flags from config toml
+    if cl_args.no_config_toml:
+        final_args = cl_args
+    else:
+        crate = cl_args.crate or cl_args.crate_flag or "."
+        final_args = get_combined_args(cl_args.config_toml, crate, cl_args)
+        validate(final_args)
+
+    post_process(final_args)
+    return final_args
+
+# Try to load Cargo.toml from dir, and update args
+def load_toml(path):
+    # Load the flag data from toml
+    data = toml.load(path)
+    rmc_data = data["rmc"]
+    flag_data = rmc_data["flags"]
+
+    # Extract nested flags
+    flags = dict()
+    def find_flags(map):
+        for key in map:
+            if type(map[key]) == dict:
+                find_flags(map[key])
+            else:
+                flags[key] = map[key]
+    find_flags(flag_data)
+
+    # Add flags to args
+    success = True
+    args = []
+    def add_flag(flag, value):
+        if type(value) == bool:
```

Comment:
What do you mean by this? The `type(value) == bool` part?

Reply:
I mean the `--no` flag handling. Is there a standard python way of handling that (e.g. an argument to the parser), or do you need to do that manually?

Reply:
The `--no` flag handling is done by a particular class of `argparser`, and it's pretty rudimentary: https://github.com/python/cpython/blob/cc602a71dfe3b3af615e78b2ca4f213d98448d9e/Lib/argparse.py#L858

Reply:
It essentially recognizes when the flag passed in starts with `--no`, and sets the value to false instead of true in that case. Since I'm constructing the arg list (rather than directly working with the parser object), I can't think of a way of avoiding manual string manipulation here.